modules/entity_version_history/tests/FunctionalJavascript/HistoryTabSettingsTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Drupal\Tests\entity_version_history\Functional;
+
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\Tests\SchemaCheckTestTrait;
+
+/**
+ * Ensures the entity version history config is correctly saved.
+ *
+ * @group entity_version_history
+ */
+class HistoryTabSettingsTest extends WebDriverTestBase {
+
+  use SchemaCheckTestTrait;
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = [
+    'node',
+    'entity_version',
+    'entity_version_workflows',
+    'entity_version_workflows_example',
+    'entity_version_history',
+    'menu_link_content',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * A user with administrative permissions.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $adminUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Create user.
+    $this->adminUser = $this->drupalCreateUser([
+      'access entity version history configuration',
+      'access administration pages',
+    ]);
+    $this->drupalLogin($this->adminUser);
+  }
+
+  /**
+   * Tests whether the history settings form is correctly saving the settings.
+   */
+  public function testHistoryTabSettingsForm() {
```

Comment:
Test is missing:

* Loading the config(s) and asserting values that the correct configs have been created/updated/delete
* More than 1 bundle and entity type that uses a version field to ensure that your form works correctly
* You can move the config schema assertions in the same test, no need for a new test just for that.